layersvt/device_simulation.cpp

```diff
@@ -103,26 +144,57 @@ std::string GetEnvarValue(const char *name) {
     return value;
 }
 
+#if defined(__ANDROID__)
+void AndroidPrintf(const char *level, const char *fmt, va_list args) {
```

Comment:
As @cnorthrop mentioned above, perhaps this could also go into a common location for Android utilities.

I'm open to refactoring the debug and error message reporting interfaces, if it would reduce the need for #if  conditionals below.